group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -3852,6 +3850,38 @@ public static String consumerGroupRebalanceTimeoutKey(String groupId, String mem
         return "rebalance-timeout-" + groupId + "-" + memberId;
     }
 
+    public static final String CLASSIC_GROUP_SIZE_COUNTER_KEY = "classic-group-size-counter";
+
+    /**
+     * Schedules (or reschedules) the group size counter for the classic groups.
+     */
+    private void scheduleClassicGroupSizeCounter() {
+        timer.schedule(
+            CLASSIC_GROUP_SIZE_COUNTER_KEY,
+            GroupCoordinatorMetricsShard.DEFAULT_GROUP_GAUGES_UPDATE_INTERVAL_MS,
+            TimeUnit.MILLISECONDS,
+            true,
+            () -> {
+                Map<ClassicGroupState, AtomicLong> groupSizeCounter = GroupCoordinatorMetricsShard.createClassicGroupGauge();
+                groups.forEach((__, group) -> {
+                    if (group.type() == CLASSIC) {
+                        groupSizeCounter.get(((ClassicGroup) group).currentState()).incrementAndGet();
+                    }
+                });
+                metrics.setClassicGroupGauges(groupSizeCounter);
```

Comment:
I wonder if we should simplify this a bit further. What do you think about the following?

```
Map<ClassicGroupState, Long> groupSizeCounter = new HashMap<>();
groups.forEach((__, group) -> {
    if (group.type() == CLASSIC) {
        groupSizeCounter.compute(((ClassicGroup) group).currentState(), Utils::incValue);
    }
});
metrics.setClassicGroupGauges(groupSizeCounter);
```